src/builders.rs

```diff
@@ -1184,103 +1029,17 @@ impl fmt::Debug for Dispatch {
                 "format",
                 &self.format.as_ref().map(|_| "<formatter closure>"),
             )
-            .field("children", &self.children)
+            .field("children_count", &self.children.len())
```

Comment:
Hmm. Wondering if there's a simple way we could keep this functionality? It probably isn't too useful for production code, but I imagine printing the full logger could be useful when debugging complicated fern setups.

I don't think it should be _too_ much of a burden to require `Debug`, but then again it would definitely mean requiring double-indirection if we ever want to accept `Box<dyn Log>`.

Reply:
Requiring `Debug` indeed isn't a hard constraint. The problem is that Rust is really difficult when working with trait objects, especially when multiple traits are involved. There simply is no such thing as `dyn Log + Debug`. I could build a custom trait, but that would lead me back to `dyn LogExt` which cannot be cast to `dyn Log`.

I'll see if I can come up with an external solution that isn't too expensive, but no guarantees

Reply:
Okay, I can propose you to try the following:

- Attach a type information string to `Output` that gets filled in when constructed
- When using general `Log` implementations instead of the provided methods, no information can be provided
- The `Dispatch` struct contains compressed information of all its children
- The `log_impl::Dispatch` does not. I do not want to keep this information after initialization
- Especially, since a nested `Dispatch` gets converted into a `log_impl::Dispatch`, this means that you'll only get the full filter information for the outermost level, and only coarse structure information for inner dispatchers.